AWSIoT/AWSIoTDataManager.m

```diff
@@ -1208,6 +1225,20 @@ - (BOOL) operationWithShadow:(NSString *)name
 - (BOOL) registerWithShadow:(NSString *)name
                     options:(NSDictionary *)options
               eventCallback:(void(^)(NSString *name, AWSIoTShadowOperationType operation, AWSIoTShadowOperationStatusType status, NSString *clientToken, NSData *payload))callback {
+    return [self registerWithShadow:name options:options eventCallback:callback completionHandler:nil];
+}
+
+- (BOOL) registerWithShadow:(NSString *)name
+                    options:(NSDictionary *)options
+              eventCallback:(void(^)(NSString *name, AWSIoTShadowOperationType operation, AWSIoTShadowOperationStatusType status, NSString *clientToken, NSData *payload))callback
+         completionCallback:(void(^)(void))completionCallback {
+    return [self registerWithShadow:name options:options eventCallback:callback completionHandler:completionCallback];
+}
+
+- (BOOL) registerWithShadow:(NSString *)name
+                    options:(NSDictionary *)options
+              eventCallback:(void(^)(NSString *name, AWSIoTShadowOperationType operation, AWSIoTShadowOperationStatusType status, NSString *clientToken, NSData *payload))callback
+          completionHandler:(nullable void(^)(void))completionHandler {
```

Comment:
Having the `nullable` annotation in the interface method resulted in weird stuff when the method was translated to Swift and autocompletion was attempted.

While not an error itself, I rather just do this to avoid confusing consumers, i.e. having the `completionCallback` declared as `nonnull` and having an implementation-only method that can take `nil`.

The same applies for the `unregisterFromShadow` method

Reply:
nit: can we rename the parameter to something with *callback to maintain consistency? something like `completionCallbackNullable`?

Reply:
Umm I think that's a very long and over-descriptive name. The `-Nullable` suffix is redundant because the parameter is already marked as `nullable`.

Since this is basically a private method, and the public API does name the parameter just `completionCallback`, I'd prefer to keep the `completionHandler` name even if it breaks consistency 😅